### DISCUSSION TITLE
What happens to empty unused topics?

### DISCUSSION THREAD

**Discussion by omidh28** [vernemq/vernemq#2015] [Q&A] [answered]
Suppose a topic has no subscriber/publish and no stored message. Does it get removed by some kind of garbage cleaner after a while or it will stay?

**Comment by ioolkos** [comment, marked as answer]
@omidh28 
Short answer: yes, topics will get deleted.
Long(er) answer: topics do not "exist" related to publish (a PUBLISH only has a topic as a routing key). Topics do get stored for Subscriber IDs, though. If a Subscriber UNSUBSCRIBES from a topic, it will be deleted/tombstoned from that SubscriberID. Note that topics also exist in RAM as a topic trie so that VerneMQ can do the reverse procedure of matching a routing key to the SubscriberIDs. 

Let me know if that answers your question. Do you feel you see something happen that should not happen?

**Reply by omidh28** [reply]
Actually I wanted to be sure that this happens, thank you.
